examples_tests/09.ColorSpaceTest/ApplicationHandler.cpp

```diff
@@ -0,0 +1,230 @@
+			break;
+
+		driver->beginScene(true, true);
+
+		driver->bindGraphicsPipeline(currentGpuPipeline.get());
+		driver->bindDescriptorSets(EPBP_GRAPHICS, currentGpuPipeline->getLayout(), 3u, 1u, &samplerDescriptorSet3.get(), nullptr);
+		driver->drawMeshBuffer(currentGpuMeshBuffer.get());
+
+		driver->blitRenderTargets(nullptr, screenShotFrameBuffer, false, false);
+		driver->endScene();
+	}
+
+	ext::ScreenShot::createScreenShoot(device, screenShotFrameBuffer->getAttachment(video::EFAP_COLOR_ATTACHMENT0), "screenShot_" + currentHandledImageFileName + ".png");
+}
+
+void ApplicationHandler::performImageTest(std::string path)
+{
+	os::Printer::log("Reading", path);
+
+	auto assetManager = device->getAssetManager();
+
+	smart_refctd_ptr<ICPUImageView> cpuImageView;
+
+	auto cpuTexture = assetManager->getAsset(path, {});
+	auto cpuTextureContents = cpuTexture.getContents();
+	
+	if (cpuTextureContents.first == cpuTextureContents.second)
+	{
+		os::Printer::log("CANNOT PERFORM THE TEST - SKIPPING. LOADING WENT WRONG", ELL_WARNING);
+		return;
+	}
+
+	io::path filename, extension, finalFileNameWithExtension;
+	core::splitFilename(path.c_str(), nullptr, &filename, &extension);
+	finalFileNameWithExtension = filename + ".";
+	finalFileNameWithExtension += extension;
+
+	// Can't do that because write is still broken
+	bool writeable = (extension != "dds");
+	smart_refctd_ptr<ICPUImageView> copyImageView;
+
+	auto asset = *cpuTextureContents.first;
+	core::smart_refctd_ptr<video::IGPUImageView> gpuImageView;
+	switch (asset->getAssetType())
+	{
+		case IAsset::ET_IMAGE:
+		{
+			ICPUImageView::SCreationParams viewParams;
+			viewParams.flags = static_cast<ICPUImageView::E_CREATE_FLAGS>(0u);
+			viewParams.image = core::smart_refctd_ptr_static_cast<asset::ICPUImage>(asset);
+			viewParams.format = viewParams.image->getCreationParameters().format;
+			viewParams.viewType = IImageView<ICPUImage>::ET_2D;
+			viewParams.subresourceRange.baseArrayLayer = 0u;
+			viewParams.subresourceRange.layerCount = 1u;
+			viewParams.subresourceRange.baseMipLevel = 0u;
+			viewParams.subresourceRange.levelCount = 1u;
+
+			cpuImageView = ICPUImageView::create(std::move(viewParams));
+		} break;
+
+		case IAsset::ET_IMAGE_VIEW:
+		{
+			cpuImageView = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(asset);
+		} break;
+
+		default:
+		{
+			os::Printer::log("EXPECTED IMAGE ASSET TYPE!", ELL_ERROR);
+			break;
+		}
+	}
+
+	copyImageView = core::smart_refctd_ptr_static_cast<ICPUImageView>(cpuImageView->clone());
+	gpuImageView = driver->getGPUObjectsFromAssets(&cpuImageView.get(), &cpuImageView.get() + 1u)->front();
+
+	if (gpuImageView)
+	{
+		auto gpuViewParams = gpuImageView->getCreationParameters();
+		gpuViewParams.image = driver->createDeviceLocalGPUImageOnDedMem(video::IGPUImage::SCreationParams(gpuViewParams.image->getCreationParameters()));
+		gpuViewParams.viewType = IGPUImageView::ET_2D;
```

Comment:
what are you doing here, this is a dead variable!

Reply:
its still dead, all 3 lines

Reply:
ah you're right